compiler/env/defines.h

```diff
@@ -136,6 +139,9 @@
 #elif defined(__aarch64__)
 #  define HOST_ARCH ARCH_ARM64
 # define TR_HOST_64BIT 1
+#elif defined(__riscv) && defined(J9RISCV64)
```

Comment:
We shouldn't test a J9 flag in OMR. Maybe testing for  `__64BIT__`, as above, would be better?

Reply:
`__64BIT__ `doesn't exist in the prefined macro list in GCC compiler on RISC-V.
`OMR_ENV_DATA64 `might be an alternative but there is no fundamental difference as compared to J9RISCV64. To keep OMR independent from OpenJ9, I agree to change to `OMR_ENV_DATA64 `.

Reply:
cmake/modules/platform/arch/riscv.cmake already defined (by one of OMR/JIT external contributor on RISCV-V)
```
if(OMR_ENV_DATA64)
	list(APPEND TR_COMPILE_DEFINITIONS -DTR_HOST_64BIT -DTR_TARGET_64BIT -DBITVECTOR_64BIT)

	set(TR_HOST_SUBARCH riscv64)
	set(TR_HOST_BITS    64)
else()
``
but they are in cmake and doesn't work in our case. We are currently still using UMA to compile the build.

Reply:
Already changed to `OMR_ENV_DATA64`.

Reply:
Unfortunately, `OMR_ENV_DATA64` is defined in `omrcfg.h`, which this header doesn't use (although, I don't know why it's this way). Maybe you can use `__SIZEOF_POINTER__ == 8`.

Reply:
Then we can just change it to `RISCV64`.

Reply:
`RISC64` is passed in as global flags on the command line.

The corresponding changes are included in `omrmakefiles/rules.linux.mk` at https://github.com/eclipse/omr/pull/4433
```
...
else ifeq (riscv,$(OMR_HOST_ARCH))
        ifeq (1,$(OMR_ENV_DATA64))
             GLOBAL_CFLAGS  +=-DRISCV64 -Wno-error=stringop-truncation -Wno-error=stringop-overflow
             GLOBAL_CXXFLAGS+=-DRISCV64 -Wno-error=stringop-truncation -Wno-error=stringop-overflow
             GLOBAL_CPPFLAGS+=-DRISCV64 -Wno-error=stringop-truncation -Wno-error=stringop-overflow
        else
```